src/app/chat/views/messages.nim

```diff
@@ -221,7 +239,7 @@ QtObject:
         let isEdit = msg.editedAt != "0" or msg.contentType == ContentType.Edit
         if not channel.muted and not isEdit and not isGroupSelf and not isMyInvite:
           let isAddedContact = channel.chatType.isOneToOne and self.isAddedContact(channel.id)
-          self.messageNotificationPushed(msg.id, channel.communityId, msg.chatId, escape_html(msg.text), msg.contentType.int, channel.chatType.int, msg.timestamp, msg.identicon, msg.userName, msg.hasMention, isAddedContact, channel.name)
+          self.messageNotificationPushed(msg.id, channel.communityId, msg.chatId, self.replacePubKeysWithMentions(msg.text, self.status.chat.contacts), msg.contentType.int, channel.chatType.int, msg.timestamp, msg.identicon, msg.userName, msg.hasMention, isAddedContact, channel.name)
```

Comment:
Passing self.status.chat.contacts is not needed to be pass as a params, you can access it directly where you use it?

Also is it working for communities, the reason for my question is that I am not sure that all members of a communities are contacts

Reply:
code needs to be removed. so if the person is not a contact the alias is returned

Reply:
see this PR: https://github.com/status-im/status-desktop/pull/3555/files we also replace for non contact. In a community you can mention someone you are not a contact with

Reply:
> This is the function we use currently, in case of communities and when the person is not a contact, we generate the alias and show that instead. and for that the self.status.chat.contacts is not needed, we simply use the public key to get the alias


Gotcha thanks for explaining ! ;)